packages/api/src/routes/medical/internal-facility.ts

```diff
@@ -26,32 +33,59 @@ router.put(
   requestLogger,
   asyncHandler(async (req: Request, res: Response) => {
     const cxId = getUUIDFrom("query", req, "cxId").orFail();
-    const facilityInput = facilityOboDetailsSchema.parse(req.body);
+    await verifyCxItVendorAccess(cxId);
 
-    const facilityUpdate: FacilityRegister = {
-      id: facilityInput.id,
+    const facilityDetails = facilityOboDetailsSchema.parse(req.body);
+    const facilityCreate: FacilityCreate = {
       cxId,
-      cqActive: facilityInput.cqActive,
-      cqType: facilityInput.cqType,
-      cqOboOid: facilityInput.cqOboOid,
-      cwActive: facilityInput.cwActive,
-      cwType: facilityInput.cwType,
-      cwOboOid: facilityInput.cwOboOid,
-      cwFacilityName: facilityInput.cwFacilityName,
       data: {
-        name: facilityInput.nameInMetriport,
-        npi: facilityInput.npi,
+        name: facilityDetails.nameInMetriport,
+        npi: facilityDetails.npi,
         address: {
-          addressLine1: facilityInput.addressLine1,
-          city: facilityInput.city,
-          state: facilityInput.state,
-          zip: facilityInput.zip,
-          country: facilityInput.country,
+          addressLine1: facilityDetails.addressLine1,
+          addressLine2: facilityDetails.addressLine2,
+          city: facilityDetails.city,
+          state: facilityDetails.state,
+          zip: facilityDetails.zip,
+          country: facilityDetails.country,
         },
       },
+      cqType: facilityDetails.cqType,
+      cwType: facilityDetails.cwType,
+      cqActive: facilityDetails.cqActive,
+      cwActive: facilityDetails.cwActive,
+      cqOboOid: facilityDetails.cqOboOid,
+      cwOboOid: facilityDetails.cwOboOid,
     };
-
-    const facility = await registerFacilityWithinHIEs(cxId, facilityUpdate);
+    let facility: Facility;
+    if (facilityDetails.id) {
+      const id = facilityDetails.id;
+      const currentFacility = await getFacilityOrFail({ cxId, id });
+      if (currentFacility.data.npi !== facilityDetails.npi) {
+        throw new Error("Cannot updated NPI once the faciilty is created on the HIEs");
+      }
+      facility = await updateFacility({ id, ...facilityCreate });
+    } else {
+      facility = await createFacility(facilityCreate);
+    }
+    const org = await getOrganizationOrFail({ cxId });
+    // TODO Move to external/hie
```

Comment:
let's tie any TODOs in the code to a specific issue number

Reply:
Done.